src/app/AttributeAccessInterface.h

```diff
@@ -88,18 +211,23 @@ class AttributeValueEncoder : protected TagBoundEncoder
      */
     FabricIndex AccessingFabricIndex() const { return mAccessingFabricIndex; }
 
-    // For consumers that can't just do a single Encode call for some reason
-    // (e.g. they're encoding a list a bit at a time).
-    TLV::TLVWriter * PrepareManualEncode()
-    {
-        // If this is called, the consumer is trying to encode a value.
-        mTriedEncode = true;
-        return mWriter;
-    }
+    AttributeEncodeState GetState() { return mEncodeState; }
```

Comment:
This is returning by value, not by reference, on purpose?  And is not a const method on purpose?